### PR TITLE
Expose ClusterMaintenenceConfig options to terraform

### PR DESCRIPTION
**Pull request by lxea**
Implements: https://github.com/gravitational/teleport/issues/26066

### CODE REVIEW

**Review by Joerger — APPROVED** [review]

File: terraform/reference.mdx
Comment (r0mant) [inline terraform/reference.mdx:1]: Shouldn't the reference documentation have been updated to include the new resource as well? @marcoandredinis

**Review by r0mant — COMMENTED** [review]
Generally looks good to me but would be good to get @marcoandredinis's review as well.

**Review by tcsc — CHANGES_REQUESTED** [review]
This doesn't build cleanly for me:
```
❯ make build
rm -rf example/terraform.tfstate
rm -rf example/terraform.tfstate.backup
rm -rf example/.terraform
rm -rf example/.terraform.lock.hcl
rm -rf ~/.terraform.d/plugins/terraform.releases.teleport.dev/gravitational/teleport/13.3.1/darwin_arm64/*
rm -rf build/*
rm -rf terraform-provider-teleport-v13.3.1-darwin-arm64-bin.tar.gz
go clean
GOOS=darwin GOARCH=arm64 CGO_ENABLED=0 go build -o build/terraform-provider-teleport  -ldflags '-w -s'
# github.com/gravitational/teleport-plugins/terraform/tfschema
tfschema/types_terraform.go:16444:24: obj.Verbs undefined (type *"github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
tfschema/types_terraform.go:16455:27: obj.Verbs undefined (type *"github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
tfschema/types_terraform.go:18089:24: obj.Verbs undefined (type *"github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
tfschema/types_terraform.go:18100:27: obj.Verbs undefined (type *"github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
tfschema/types_terraform.go:21881:106: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
tfschema/types_terraform.go:21886:106: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
tfschema/types_terraform.go:21889:26: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
tfschema/types_terraform.go:21891:31: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
tfschema/types_terraform.go:21892:106: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
tfschema/types_terraform.go:21894:42: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
tfschema/types_terraform.go:21894:42: too many errors
make: *** [build] Error 1
```

File: terraform/reference.mdx
Comment (marcoandredinis) [reply terraform/reference.mdx:1]: New resources must be added to `terraform/gen/main.go`
```go
// Create Docs Markdown
var (
	mapResourceSchema = map[string]func(context.Context) (tfsdk.Schema, diag.Diagnostics){
		"cluster_maintenance_config": tfschema.GenSchemaClusterMaintenanceConfigV1,
		// ...
```

**Comment by lxea** [comment]
> 



> This doesn't build cleanly for me:
> 
> ```
> ❯ make build
> rm -rf example/terraform.tfstate
> rm -rf example/terraform.tfstate.backup
> rm -rf example/.terraform
> rm -rf example/.terraform.lock.hcl
> rm -rf ~/.terraform.d/plugins/terraform.releases.teleport.dev/gravitational/teleport/13.3.1/darwin_arm64/*
> rm -rf build/*
> rm -rf terraform-provider-teleport-v13.3.1-darwin-arm64-bin.tar.gz
> go clean
> GOOS=darwin GOARCH=arm64 CGO_ENABLED=0 go build -o build/terraform-provider-teleport  -ldflags '-w -s'
> # github.com/gravitational/teleport-plugins/terraform/tfschema
> tfschema/types_terraform.go:16444:24: obj.Verbs undefined (type *"github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
> tfschema/types_terraform.go:16455:27: obj.Verbs undefined (type *"github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
> tfschema/types_terraform.go:18089:24: obj.Verbs undefined (type *"github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
> tfschema/types_terraform.go:18100:27: obj.Verbs undefined (type *"github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
> tfschema/types_terraform.go:21881:106: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
> tfschema/types_terraform.go:21886:106: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
> tfschema/types_terraform.go:21889:26: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
> tfschema/types_terraform.go:21891:31: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
> tfschema/types_terraform.go:21892:106: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
> tfschema/types_terraform.go:21894:42: obj.Verbs undefined (type "github.com/gravitational/teleport/api/types".KubernetesResource has no field or method Verbs)
> tfschema/types_terraform.go:21894:42: too many errors
> make: *** [build] Error 1
> ```

I encountered this issue at some point also, I can't remember how I resolved it, but I think it might be to do with the teleport commit in the go.mod and makefile where it `go mod download`s teleport, it also wont built until #30214 is merged

**Comment by marcoandredinis** [comment]
This looks mostly good :+1: 

> it also wont built until https://github.com/gravitational/teleport/pull/30214 is merged

Not only merged, but also backported to v13 (teleport-plugins follows only the latest major teleport version)

**Review by tcsc — APPROVED** [review]
Looks good to me once gravitational/teleport#30214 and the subsequent v13 backport are merged.

**Comment by marcoandredinis** [comment]
Let me know if I should do another pass.
It looks like the CI is failing.

**Pull request merged**